### PR TITLE
[Core] do nothing

### PR DESCRIPTION
**Pull request by otocephaly**
added core notcasting based on BLM's version. moved BLM's version to core and added AST

## Pull request type

- [x] This is new functionality or an addition to existing functionality

## Pull request details

- [x] This is in response to a discussion or thread on Discord: *[Link to thread or start of discussion]*(https://discord.com/channels/441414116914233364/1285154582292332554/1285154582292332554)
- [x] The goal of this PR is detailed below:
    - migrate BLM's notcasting to core and apply to all casters

## Testing / Validation

- [x] I used the log(s) listed below to develop and test this bugfix:
    - https://www.fflogs.com/reports/kCbDJx6gVKzmfMZa#fight=6&source=71  (BLM)
    - https://www.fflogs.com/reports/2a7mFHPM86yARbjV#fight=1&source=21 (AST)
    - https://www.fflogs.com/reports/GJ7XBjQ3q4aWADZ6#fight=8&source=54 (BLM high parse)

## Job Maintenance
- [x] I have collaborated with one or more job maintainers for the job(s) in this PR while developing it

### CODE REVIEW

File: src/data/ACTIONS/root/PCT.ts [hunk 1]
```diff
@@ -88,6 +88,7 @@ export const PCT = ensureActions({
 		onGcd: true,
 		speedAttribute: Attribute.SPELL_SPEED,
 		castTime: 2300,
+		gcdRecast: 3300,
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/PCT.ts:91]: Having a gcdRecast on the actions ends up implying that they have an additional cooldown like Phlegma, so they get duplicated on the timeline. These changes should be removed and ABC/NotCasting should treat cooldown as gcdRecast if gcdRecast is not present (other modules have that behavior too, even if it's a little wonky)

File: src/parser/core/modules/NotCasting.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+interface Window {
+	start: number,
+	stop?: number,
+	recastTime: number,
+}
+
+export class NotCasting extends Analyser {
+	static override handle = 'notcasting'
+	static override title = t('core.notcasting.title')`Times you did literally nothing`
+
```
Comment (AkairyuGestalter) [inline src/parser/core/modules/NotCasting.tsx:28]: While this was a BLM-only thing, I wasn't as worried about this title, but if we're proposing making it general to all casting jobs, we really need a better way to label this...

File: src/parser/core/modules/NotCasting.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+		this.addEventHook({
+			type: 'death',
+			actor: this.parser.actor.id,
+		}, this.onDeath)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: Events['action']) {
+		//some jobs might care about oGCDs. e.g. BLM
+		const actionIsOGCD: boolean = !this.data.getAction(event.action)?.onGcd ?? false
```
Comment (AkairyuGestalter) [inline src/parser/core/modules/NotCasting.tsx:55]: This line's giving me a "right operand of ?? is unreachable" error. Either isn't necessary, or need to rewrite it to properly trap the null

File: src/parser/core/modules/NotCasting.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+		}
+		//this cast is our new last cast
+		this.noCastWindows.current = {
+			start: timeStamp,
+			recastTime: actionRecast,
+		}
+	}
+
+	private onBegin(event: Events['prepare']) {
+		const actionIsOGCD: boolean = !this.data.getAction(event.action)?.onGcd ?? false
```
Comment (AkairyuGestalter) [inline src/parser/core/modules/NotCasting.tsx:89]: Same comment here about the nullish check maybe not being necessary

**Review by AkairyuGestalter — CHANGES_REQUESTED** [review]

File: src/data/ACTIONS/root/PCT.ts (hunk 1 above)
Comment (otocephaly) [reply src/data/ACTIONS/root/PCT.ts:91]: ah okie that makes sense! I'll do that (I must've been over thinking this a lil)
Thank you!

**Comment by otocephaly** [comment]
retiring this PR in favour of PR https://github.com/xivanalysis/xivanalysis/pull/2114
will revisit this format or code if necessary

**Pull request closed without merging**